docs/oathkeeper/pipeline/mutator.md

```diff
@@ -476,6 +477,10 @@ Example request/response payload:
   }
 }
 ```
+:::warning
+Ory Oathkeeper expects custom headers followed by canonical header notation. Every header should start with the uppercase first letter and any other letter following a hyphen. For example, `x-user` should be passed to oathkeeper as `X-User`
+:::
```

Comment:
Hey man, I have a few problems with this warning. I think I know what you mean but I'm not sure.

- Oathkeeper needs to get custom headers with names that follow this format `UPPERCASE-UPPERCASElowercase`right? 
- Is there such a thing as "canonical HTTP headers" that defines the format ? When I google that I get articles about `rel=”canonical”` headers.

Reply:
- UPPERCASE-UPPERCASElowercase-UPPERCASE-lowercase
- This is Go specific thing and related to the Go programming language. [Header](https://pkg.go.dev/net/http#Header) documentation and [textproto](https://pkg.go.dev/net/textproto#CanonicalMIMEHeaderKey) documentation

Reply:
Do all custom headers have to start with `X-`? 

So the rule is `X-[Uppercase]lowercase-[Uppercase]lowercase(...)`?

Reply:
>Do all custom headers have to start with X-?
Basically yes

>So the rule is X-[Uppercase]lowercase-[Uppercase]lowercase(...)?
Yep.